common/flatpak-dir.c

```diff
@@ -8691,8 +8781,10 @@ flatpak_dir_get_all_installed_refs (FlatpakDir  *self,
         return NULL;
 
       for (i = 0; app_refs[i] != NULL; i++)
-        g_hash_table_insert (local_refs, g_strdup (app_refs[i]),
-                             GINT_TO_POINTER (1));
+        {
+          FlatpakCollectionRef *ref = flatpak_collection_ref_new (NULL, app_refs[i]);
```

Comment:
If any of these apps were installed from a remote that had a collection id, it doesn't really make sense to return a FlatpakCollectionRef here with NULL collection-id, does it? It will not match the one you would get from the remote. Not sure what or if anything this breaks, but it does make it clear that either this has to be fixed, or using a FlatpakCollectionRef here is not right.

Reply:
None of the current callers would use it but they aren't hurt by it either. And fortunately getting the collection ID is O(1) so this is still O(n). Pushed a fixup